src/main/java/com/felixgrund/codeshovel/parser/impl/TypeScriptParser.java

```diff
@@ -0,0 +1,165 @@
+package com.felixgrund.codeshovel.parser.impl;
+
+import com.eclipsesource.v8.V8;
+import com.eclipsesource.v8.V8Array;
+import com.eclipsesource.v8.V8Object;
+import com.felixgrund.codeshovel.changes.*;
+import com.felixgrund.codeshovel.entities.Ycommit;
+import com.felixgrund.codeshovel.exceptions.ParseException;
+import com.felixgrund.codeshovel.parser.AbstractParser;
+import com.felixgrund.codeshovel.parser.Yfunction;
+import com.felixgrund.codeshovel.parser.Yparser;
+import com.felixgrund.codeshovel.util.Utl;
+import com.felixgrund.codeshovel.visitors.TypeScriptVisitor;
+import com.felixgrund.codeshovel.wrappers.Commit;
+import com.felixgrund.codeshovel.wrappers.StartEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TypeScriptParser extends AbstractParser implements Yparser {
+
+    public static final String ACCEPTED_FILE_EXTENSION = ".*\\.(t|j)sx?$";
+    private Logger log = LoggerFactory.getLogger(TypeScriptParser.class);
+
+    public TypeScriptParser(StartEnvironment startEnv, String filePath, String fileContent, Commit commit) throws ParseException {
+        super(startEnv, filePath, fileContent, commit);
+    }
+
+    @Override
+    protected List<Yfunction> parseMethods() throws ParseException {
+        try {
+            TypeScriptMethodVisitor visitor = new TypeScriptMethodVisitor() {
+                @Override
+                public boolean methodMatches(Yfunction method) {
+                    return method.getBody() != null;
+                }
+            };
+            visitor.visit(this.fileContent);
+            return visitor.getMatchedNodes();
+        } catch (Exception e) {
+            throw new ParseException(e.getMessage(), this.filePath, this.fileContent);
+        }
+    }
+
+    @Override
+    public double getScopeSimilarity(Yfunction function, Yfunction compareFunction) {
+        return Utl.parentNamesMatch(function, compareFunction) ? 1.0 : 0.0; // TODO
```

Comment:
This was taken directly from the old JS parser, and am not entirely sure what it was to be used for or how it was meant to be calculated. Maybe @jyoo980 has an idea?